File: BOJ/1158.py
import sys


N, K = [int(i) for i in sys.stdin.readline().split()]
people = [i for i in range(1, N+1)]
loc = 0
answer = []
while people:
    loc += K - 1
    if loc >= len(people):
        loc %= len(people)
    answer.append(people.pop(loc))
print('<', end='')
for i in answer[:-1]:
    print(i, end=', ')
print(answer[-1], end='>')

File: BOJ/1158.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.List;

public class Main {

	private static final StringBuilder sb = new StringBuilder();

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		sb.append("<");
		String[] input = br.readLine().split(" ");
		int N = Integer.parseInt(input[0]);
		int K = Integer.parseInt(input[1]);
		List<Integer> arr = new LinkedList<>();
		int[] ans = new int[N];
		for (int i = 1; i <= N; i++) {
			arr.add(i);
		}
		int loc = 0;
		for (int i = 0; i < N; i++) {
			loc += K - 1;
			loc %= arr.size();
			ans[i] = arr.remove(loc);
		}
		sb.append(ans[0]);
		for (int i = 0; i < N - 1; i++) {
			sb.append(",").append(" ").append(ans[i+1]);
		}
		sb.append(">");
		System.out.print(sb);
	}
}